File: back-end/src/main/java/com/action/dao/NewsUserDao.java
package com.action.dao;

import com.action.entity.NewsUser;
import org.apache.ibatis.annotations.Insert;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Select;

@Mapper
public interface NewsUserDao {

    @Select("select uid, username, user_pwd as userPwd, nick_name as nickName " +
            "from news_user " +
            "where username = #{username}")
    NewsUser findByUsername(String username);

    @Select("select uid, username, user_pwd as userPwd, nick_name as nickName " +
            "from news_user " +
            "where uid = #{userId}")
    NewsUser findByUid(Integer userId);


    @Insert("insert into news_user " +
            "values (DEFAULT, #{username}, #{userPwd}, #{nickName})")
    Integer add(NewsUser user);
}

File: back-end-gin/dao/news_user_dao.go
package dao

import (
	"github.com/action/micro-headlines/models"
	"gorm.io/gorm"
)

type NewsUserDao struct {
	db *gorm.DB
}

func NewNewsUserDao(db *gorm.DB) *NewsUserDao {
	return &NewsUserDao{db: db}
}

// FindByUsername 根据用户名查询用户
func (d *NewsUserDao) FindByUsername(username string) (*models.NewsUser, error) {
	var user models.NewsUser
	err := d.db.Where("username = ?", username).First(&user).Error
	if err != nil {
		if err == gorm.ErrRecordNotFound {
			return nil, nil
		}
		return nil, err
	}
	return &user, nil
}

// FindByUid 根据用户 ID 查询用户
func (d *NewsUserDao) FindByUid(uid int) (*models.NewsUser, error) {
	var user models.NewsUser
	err := d.db.Where("uid = ?", uid).First(&user).Error
	if err != nil {
		if err == gorm.ErrRecordNotFound {
			return nil, nil
		}
		return nil, err
	}
	return &user, nil
}

// Add 添加用户
func (d *NewsUserDao) Add(user *models.NewsUser) error {
	return d.db.Create(user).Error
}

